### PR TITLE
REF: Avoid np.can_cast for scalar inference for NEP 50

### PR DESCRIPTION
**Pull request by mroeschke**
Looks like using `np.can_cast` for scalar casting inference is deprecated. We should probably use our own scalar casting inference routines instead

### CODE REVIEW

File: pandas/core/dtypes/cast.py [hunk 1]
```diff
@@ -699,7 +699,9 @@ def _maybe_promote(dtype: np.dtype, fill_value=np.nan):
             dtype = np.dtype(np.object_)
 
         elif issubclass(dtype.type, np.integer):
-            if not np.can_cast(fill_value, dtype):
+            try:
+                np_can_hold_element(dtype, fill_value)
```
Comment (lithomas1) [inline pandas/core/dtypes/cast.py:703]: Maybe we should right a wrapper function for this to make our own can cast.

Comment (mroeschke) [reply pandas/core/dtypes/cast.py]: Good idea. Wrapped in a helper function

**Comment by mroeschke** [comment]
Looks like the test failures are timeouts hopefully unrelated to this change. Going to backport this change as this could be useful for 2.1.x

**Comment by pllim** [comment]
Thank you! When do you plan to push out a new dev wheel with this patch? 🙏

**Comment by mroeschke** [comment]
Wheels uploads run nightly at 3:27 UTC, so hopefully the dev wheels should be available tomorrow

**Comment by pllim** [comment]
I can confirm the dev wheel is up and it works now. Thanks again!

**Pull request merged**